R/site-calade.R

```diff
@@ -0,0 +1,230 @@
+  usethis::ui_done("Patching example site")
+  calade_write_hugodown(path)
+  calade_write_sentinel(path)
+  calade_write_readme(path)
+  calade_write_css(path)
+  calade_rename_default_archetype(path)
+  calade_patch_rmd_dir(path(path, "themes", "calade", "archetypes"))
+  calade_patch_rmd_dir(path(path, "content"))
+  calade_patch_head_custom(path)
+  calade_patch_config(path)
+
+  # Build rmd posts/projects to hugo-flavoured md and then build
+  if(knit == TRUE) {
+    usethis::ui_done("Knitting .Rmd files to .md")
+    lapply(site_outdated(site = path), calade_build_post)
+  }
+
+  # Open in a new session if requested
+  if (open) {
+    usethis::proj_activate(path)
+  }
+}
+
+
+
+# helpers -----------------------------------------------------------------
+
+calade_build_post <- function(path) {
+  split_path <- path_split(path)[[1]]
+  local_root <- which(split_path == "content")
+  tidy_path <- path_join(split_path[local_root:(length(split_path))])
+  usethis::ui_line(paste("    ", tidy_path))
+  suppressWarnings(rmarkdown::render(path, quiet = TRUE))
+}
+
+
+# Downloads and extracts the Hugo theme
+calade_download_theme <- function() {
+  zip <- curl::curl_download(
+    "https://github.com/djnavarro/hugo-calade/archive/master.zip",
+    file_temp("hugodown")
+  )
+  exdir <- file_temp("hugodown")
+  utils::unzip(zip, exdir = exdir)
+  exdir <- path(exdir, "hugo-calade-master")
+  return(exdir)
+}
+
+
+# Convert md post archetypes to Rmd
+calade_rename_default_archetype <- function(path) {
+  dir_path <- path(path, "themes", "calade", "archetypes")
+  new_path <- path(dir_path, "default.Rmd")
+  file_move(
+    path = path(dir_path, "default.md"),
+    new_path = new_path
+  )
+}
+
+
+# Patch the yaml header in an rmd file
+calade_patch_rmd <- function(path) {
+  lines <- brio::read_lines(path)
+  lines <- c(lines[1],
+             "output: hugodown::md_document",
+             lines[-1]
+  )
+  brio::write_lines(lines, path)
+}
+
+
+# Patch all rmd files in a folder
+calade_patch_rmd_dir <- function(path) {
+  rmd_files <- dir_ls(path = path, glob = "*.Rmd", recurse = TRUE)
+  lapply(rmd_files, calade_patch_rmd)
+}
+
+
+# Writes the hugodown yaml file
+calade_write_hugodown <- function(path) {
+  opts <- list(hugo_version = "0.66.0")
+  yaml::write_yaml(opts, path(path, "_hugodown.yaml"))
+}
+
+
+# Writes the readme file
+calade_write_readme <- function(path) {
+  file_copy(path_package("hugodown", "calade", "README.md"), path)
+}
+
+
+# Writes the sentinel file
+calade_write_sentinel <- function(path) {
+  file_copy(path_package("hugodown", "calade", "index.Rmd"), path)
+}
+
+
+# Copies the highlight.css style file across
+calade_write_css <- function(path) {
+  dir_create(path(path, "static", "css"))
+  file_copy(
+    path = path_package("hugodown", "calade", "highlight.css"),
+    new_path = path(path, "static", "css")
+  )
+}
+
+
+# Inserts link to highlight.css file in head_custom.html
+calade_patch_head_custom <- function(path) {
```

Comment:
Could this be done in calade directly? You'd also need to do it if you were to use with blogdown.